File: core/src/main/java/com/triangleleft/flashcards/ui/main/IDrawerView.java
package com.triangleleft.flashcards.ui.main;

import com.triangleleft.flashcards.service.settings.Language;
import com.triangleleft.flashcards.ui.common.view.IView;

import java.util.List;

public interface IDrawerView extends IView {

    void showUserData(String username, String avatar, List<Language> languages);

    void notifyLanguageSwitchError();

    void showDrawerProgress();

}

File: core/src/main/kotlin/com/triangleleft/flashcards/ui/main/IDrawerView.kt
package com.triangleleft.flashcards.ui.main

import com.triangleleft.flashcards.service.settings.Language
import com.triangleleft.flashcards.ui.common.view.IView
import io.reactivex.Observable

interface IDrawerView : IView {

    fun render(state: DrawerViewState)

    fun languageClicks(): Observable<Language>

    fun errorShowCompletes(): Observable<Any>

    enum class Page {
        CONTENT,
        PROGRESS
    }
}
